cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1228,17 +1237,46 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
                             resultSts = (StatefulSet) ((ReconcileResult) result.resultAt(0)).resource();
                         }
 
-                        return kafkaSetOperations.maybeRollingUpdate(sts, pod -> {
+                        Function<Pod, List<String>> fn = pod -> {
                             log.info("{}: Downgrade: Patch + rolling update of {}: Pod {}", reconciliation, stsName, pod.getMetadata().getName());
-                            return "Downgrade phase 1 of " + phases + ": Patch + rolling update of " + name + ": Pod " + pod.getMetadata().getName();
-                        }).map(resultSts);
+                            return singletonList("Downgrade phase 1 of " + phases + ": Patch + rolling update of " + name + ": Pod " + pod.getMetadata().getName());
+                        };
+                        return maybeKafkaRoll(sts, fn).map(resultSts);
                     })
                     .compose(ss2 -> {
                         log.info("{}: {}, phase 1 of {} completed", reconciliation, versionChange, phases);
                         return Future.succeededFuture(ss2);
                     });
         }
 
+        protected CompositeFuture adminClientSecrets() {
+            Future<Secret> clusterCaCertSecretFuture = secretOperations.getAsync(
+                namespace, KafkaResources.clusterCaCertificateSecretName(name)).compose(secret -> {
+                    if (secret == null) {
+                        return Future.failedFuture(Util.missingSecretException(namespace, KafkaCluster.clusterCaCertSecretName(name)));
+                    } else {
+                        return Future.succeededFuture(secret);
+                    }
+                });
+            Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                namespace, ClusterOperator.secretName(name)).compose(secret -> {
+                    if (secret == null) {
+                        return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(name)));
+                    } else {
+                        return Future.succeededFuture(secret);
+                    }
+                });
+            return CompositeFuture.join(clusterCaCertSecretFuture, coKeySecretFuture);
+        }
+
+        Future<Void> maybeKafkaRoll(StatefulSet sts, Function<Pod, List<String>> podNeedsRestart) {
```

Comment:
```suggestion
        Future<Void> maybeRollKafka(StatefulSet sts, Function<Pod, List<String>> podNeedsRestart) {
```
Small suggested name change, a doc comment explaining why this is so highly configurable would be useful too.